packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1025 @@
+                    this.focusState.headerActionIndex = headerActionIndex + 1;
+                    focusableElement = this.focusableHeaderElements.headerActions[
+                        this.focusState.headerActionIndex
+                    ];
+                }
+            }
+        }
+        if (focusableElement) {
+            this.inNavigationMode = this.focusState.focusType !== TableFocusType.cellContent
+                && this.focusState.focusType !== TableFocusType.cellActionMenu;
+            this.focusElement(focusableElement);
+            return true;
+        }
+        this.blurAfterLastTab(activeElement);
+        return false;
+    }
+
+    private blurAfterLastTab(activeElement: HTMLElement): void {
+        // In order to get the desired browser-provided Tab/Shift-Tab behavior of focusing the
+        // element before/after the table, the table shouldn't have tabIndex=0 when this event
+        // handling ends. However it needs to be tabIndex=0 so we can re-focus the table the next time
+        // it's tabbed to, so set tabIndex back to 0 after a rAF.
+        // Note: In Chrome this is only needed for Shift-Tab, but in Firefox both Tab+Shift-Tab need this
+        // to work as expected.
+        this.table.tabIndex = -1;
+        window.requestAnimationFrame(() => {
+            this.table.tabIndex = 0;
+        });
+
+        activeElement.blur();
+        this.setElementFocusable(activeElement, false);
+    }
+
+    private focusElement(
+        element: HTMLElement,
+        focusOptions?: FocusOptions
+    ): void {
+        const previousActiveElement = this.getActiveElement();
+        if (previousActiveElement !== element) {
+            this.setElementFocusable(element, true);
+            element.focus(focusOptions);
+            if (previousActiveElement) {
+                this.setElementFocusable(previousActiveElement, false);
+            }
+        }
+    }
+
+    private setElementFocusable(
+        element: HTMLElement,
+        focusable: boolean
+    ): void {
+        if (element === this.table) {
+            return;
+        }
+
+        const tabIndex = focusable ? 0 : -1;
+        const menuButton = element instanceof MenuButton
+            ? element
+            : this.getContainingMenuButton(element);
+        if (menuButton) {
+            menuButton.tabIndexOverride = tabIndex;
+            // The MenuButton needs to be visible in order to be focused, so this 'focused' CSS class styling
+            // handles that (see cell/styles.ts).
+            // TableCell.onActionMenuButtonBlur() ensures that the CSS class is removed when the action menu
+            // button is blurred in all cases.
+            if (focusable) {
+                menuButton.classList.add('focused');
+            } else {
+                menuButton.classList.remove('focused');
+            }
+        }
+
+        if ('tabIndexOverride' in element) {
```

Comment:
At the time that I'd added `tabIndexOverride`, setting `tabIndex` on the controls (where they're consumed from the table/row/cell templates) wasn't working, because the controls already set `tabIndex` in their own template and that was taking precedence. It was mainly added so that those controls would start out as `tabindex=-1` when used from the table, before they're focused via the keyboard.

I need to take another stab at getting everything working just via the existing `tabIndex` property though (Milan made the same comment).

Reply:
Removed all the tabIndexOverride code from this PR. Replaced with (temporary) `NotFocusableBehavior` and `getTabIndexTarget()`.

#2094 covers the tabIndex changes for this feature.